src/main/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCounterSignedCitizenActionsHandler.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.cmc.claimstore.events.offer;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.claimstore.config.properties.notifications.NotificationsProperties;
+import uk.gov.hmcts.cmc.claimstore.services.notifications.NotificationReferenceBuilder.AgreementCounterSigned;
+import uk.gov.hmcts.cmc.claimstore.services.notifications.OfferMadeNotificationService;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.offers.MadeBy;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.CLAIM_REFERENCE_NUMBER;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.COUNTER_SIGNING_PARTY;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.FRONTEND_BASE_URL;
+
+@Component
+public class AgreementCounterSignedCitizenActionsHandler {
+
+    private final OfferMadeNotificationService offerMadeNotificationService;
```

Comment:
Can you please rename this class as it has nothing to do with offers. It is regular notification service with retry. Not sure why someone created it. Makes sense to clean it up.

Name like `NotificationService` would be enough.

Would be good whether we do not have more copies as well.